packages/eslint-plugin/tests/schema-snapshots/naming-convention.shot

```diff
@@ -1572,7 +1572,6 @@ type Options = /** Selector 'default' */
       suffix?: PrefixSuffixConfig;
       trailingUnderscore?: UnderscoreOptions;
     }
-  /** Multiple selectors in one config */
```

Comment:
do we want to preserve description here?

Reply:
hmm - this is a result of my last change to use TS to automatically simplify unions.
it appears that TS does not retain the comments on union members 😢

Maybe we can't do this then and we're just stuck with uncollapsed unions sadge

Reply:
hmm wdyt about changing `printUnionType` to

```ts
function printUnionType(members: (string | GeneratedResult)[]): string {
  return members
    .map(member => {
      if (typeof member === 'string') {
        return `${member}`;
      }
      return `${printComment(member)} ${member.code}`;
    })
    .join(' | \n');
}
```

there is few differences tho:

![image](https://user-images.githubusercontent.com/625469/232870129-fa64169e-5eec-4171-a579-216dde8feeff.png)

![image](https://user-images.githubusercontent.com/625469/232870285-244f9ede-7c73-44aa-8c22-7ec17b814a9d.png)

Reply:
The big problem right now is that we are building strings - and it's really hard to tell if a string returned from the function needs to be wrapped or not - so we just always wrap it.

I think we can just do better here by rewriting the algorithm to build a pseudo-AST instead of building strings - that way we don't need to worry about superfluous parentheses because we can make decisions on the fly as to whether or not we need parentheses or how things get nested.
Can also do a deduplication pass as well.
It'll be much easier to do an optimisation pass when printing the AST from the top down as opposed to trying to optimise the strings from the bottom up.

Reply:
> I think we can just do better here by rewriting the algorithm to build a pseudo-AST

I get that, i'm just wandering how granule you want to make it, in teory something like this for now could be sufficient:

```ts
type GeneratedResult =
  | {
      readonly code: string;
      readonly type: 'raw';
      readonly commentLines?: string[];
    }
  | {
      readonly code: string[];
      readonly type: 'union';
      readonly commentLines?: string[];
    };
```

----------------

i'm not sure how high this is on your PRIO list, I really like this change :>

Reply:
yeah I wasn't going to go full ts-estree AST - just a light-weight object to do exactly what you described!

I wonder actually - I actually might just be able to remove the parentheses cos we only generate literals, arrays, objects, and unions - none of which require parentheses...

I'll play around and see.

Removing TS is good anyway cos it's a pretty slow way of doing things (makes the schema test take ~10s). But it's good to know how easy it is to use `@typescript/vfs` to do a quick typecheck like this.